server/deadfish.hpp

```diff
@@ -65,8 +81,20 @@ struct Collideable {
 	Collideable(const Collideable&) = delete;
 };
 
-struct Mob : public Collideable {
-	uint16_t mobID = 0;
+struct Movable {
+	FlatBuffGenerated::Vec2 pos;
+	uint16_t movableID;
+	float angle;
+
+	virtual std::unique_ptr<FlatBuffGenerated::MovableComponent> fbMovable();
+};
+
+struct CollideableMovable
+	: public Collideable, public Movable {
```

Comment:
It's starting to look a bit spooky :D But as long as we don't have more object properties as these two (and I don't suppose we're going to) it's ok.

Reply:
This particular combo is important. This struct overloads the function fbMovable to handle copying data from the physics engine to the movable structure